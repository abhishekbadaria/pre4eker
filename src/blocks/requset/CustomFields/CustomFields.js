import React from "react";
import style from "./CustomFields.css";

import Input from "src/blocks/Input/Input.js";

export default function CustomFields(props) {
  const { addCustomField, fields, removeCustomField } = props;

  return (
    <div className={style.cf__wrapper}>
      <span className={style.cf__btlockName}>Дополнительные поля</span>
      <button
        className={style.cf__addBtn}
        onClick={() => props.addCustomField()}
      >
        +
      </button>
      {fields &&
        fields.map((item, index) => (
          <div className={style.cfImet} key={`cf_${index}`}>
            <Input label="Идентификатор" name="externalID " />
            <Input label="Значение" name="value " />

            <button
              className={style.cf__removeBtn}
              onClick={(e) => props.removeCustomField(e)}
            >
              -
            </button>
          </div>
        ))}
    </div>
  );
}
