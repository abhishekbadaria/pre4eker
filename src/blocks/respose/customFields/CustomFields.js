import React, { useState } from "react";
import PropTypes from "prop-types";
import stylesShared from "src/customer.css";
import Button from "src/blocks/Button/Button";

const CustomFields = ({ data }) => {
  const keys = Object.keys(data);

  const [showCustomFields, toggle] = useState(false);

  const drawValues = (values) =>{
   switch (typeof values) {
     case 'object':
       return values.map((value, i) => (
        <div key={`item ${i}`}>
          {`${value}` || <span className={stylesShared.neutral}>Пусто</span>}
        </div>
      ));   
     default:
      return `${values}` || (
        <span className={stylesShared.neutral}>Пусто</span>
      ) 
   }
   }

  return (
    <div>
      <div className={stylesShared.name}>
        Дополнительные поля 
        <Button
          type="DROPDOWN"
          size="small"
          action={() => toggle(!showCustomFields)}
        />
      </div>

      {showCustomFields &&
        keys.map((field, index) => (
          <div key={`field ${index}`} className={`${stylesShared.inline} ${stylesShared.line}`}>
            <div className={`${stylesShared.half} ${stylesShared.value}`}>
              {field}
            </div>
            <div className={`${stylesShared.half} ${stylesShared.value}`}>
              {drawValues(data[field])}
            </div>
          </div>
        ))}
    </div>
  );
};

CustomFields.propTypes = {
  data: PropTypes.object,
};

export default CustomFields;
