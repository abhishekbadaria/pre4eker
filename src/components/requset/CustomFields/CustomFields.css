.inline {
  display: flex;
  margin: 0 -.3em 0 -.3em;
  flex-wrap: wrap;
  align-items: center;
  justify-content: space-between;
}

.half {
  width: 50%;
  padding: 1rem .3rem;
}

.third {
  width: 33%;
  padding: 0 .3rem;
}

.h2 {
  margin: 0;
}

.h4 {
  margin: 0;
}