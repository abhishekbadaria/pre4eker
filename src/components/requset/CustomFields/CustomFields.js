import React, { useState, useEffect } from "react";

import PropTypes from "prop-types";

import Input from "src/components/Input/Input";
import Button from "src/components/Button/Button";

import style from "./CustomFields.css";

import arrayFunctions from "../arrayFunctions";

const CustomFields = ({ body, setBody, typeOfParrent }) => {
  // let initialCustomFileds;

  const initialCustomFileds = (data) => {
    if (typeof data.customFields !== "undefined") {
      return Object.keys(data.customFields).map((customField, index) => ({
        number: index,
        field: customField,
        value: data.customFields[customField],
      }));
    }
    return [];
  };

  console.log(body);
  console.log(initialCustomFileds(body));

  const [customFields, setCustomFileds] = useState(initialCustomFileds(body));

  useEffect(() => {
    let customFieldsToState = {};
    customFields.forEach(({ field, value }) => {
      if (typeof value !== "undefined" && typeof field !== "undefined") {
        customFieldsToState = { ...customFieldsToState, [field]: value };
      }
      return false;
    });

    if (
      Object.keys(customFieldsToState).length > 0 &&
      Object.keys(customFieldsToState).length ===
        Object.keys(customFields).length
    ) {
      setBody({
        ...body,
        customFields: customFieldsToState,
      });
    }
  }, [customFields]);

  useEffect(() => {
    setCustomFileds(initialCustomFileds(body));
  }, [body]);

  return (
    <div className={style.customFields}>
      <div className={style.inline}>
        <h4 className={style.h4}>Дополнительные поля</h4>
        <Button
          action={() => setCustomFileds(arrayFunctions.addItem(customFields))}
          type="ADD"
        />
      </div>

      {customFields.map((item) => (
        <div key={item.number} className={`${style.item} ${style.inline}`}>
          <div>
            <Input
              label="Название поля"
              name={`customField-${typeOfParrent}-${item.number}-field`}
              value={item.field}
              onChange={(e) => {
                const upDatedValue = {
                  ...item,
                  field: e.target.value,
                };
                setCustomFileds(
                  arrayFunctions.updateItem(customFields, upDatedValue)
                );
              }}
            />
          </div>
          <div>
            <Input
              label="Значение поля"
              name={`customField-${typeOfParrent}-${item.number}-value`}
              value={item.value}
              onChange={(e) => {
                const upDatedValue = {
                  ...item,
                  value: e.target.value,
                };
                setCustomFileds(
                  arrayFunctions.updateItem(customFields, upDatedValue)
                );
              }}
            />
          </div>
          <Button
            action={() => {
              setCustomFileds(arrayFunctions.removeItem(customFields, item));
            }}
            type="REMOVE"
          />
        </div>
      ))}
    </div>
  );
};

CustomFields.propTypes = {
  body: PropTypes.object,
  setBody: PropTypes.func,
  typeOfParrent: PropTypes.string,
};

CustomFields.defaultProps = {
  body: {},
  setBody: () => ({}),
  typeOfParrent: "",
};

export default CustomFields;
