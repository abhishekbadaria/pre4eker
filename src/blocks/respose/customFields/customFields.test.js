import React from "react";
import CustomFields from "./CustomFields";
import { render, fireEvent, waitFor, screen, cleanup, act } from '@testing-library/react' 
import response from 'Data/response'

test(`customfields shoulg match snapshot`, () => {
  const {asFragment, getByRole} = render(<CustomFields data={response.customer.customFields} />)
  expect(asFragment(<CustomFields />)).toMatchSnapshot();

  fireEvent.click(screen.getByRole(`button`))
  expect(asFragment(<CustomFields />)).toMatchSnapshot();
  
  fireEvent.click(screen.getByRole(`button`))
  expect(asFragment(<CustomFields />)).toMatchSnapshot();

})