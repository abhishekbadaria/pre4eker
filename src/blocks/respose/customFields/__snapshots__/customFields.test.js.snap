// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customfields shoulg match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="name"
    >
      Дополнительные поля
      <button
        class="small button normal undefined basic"
      >
        <div>
          <svg
            fill="none"
            height="5"
            viewBox="0 0 8 5"
            width="8"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M0.0344238 1.16905L0.732916 0.502381L3.54531 3.1561L6.33428 0.499939L7.03473 1.16647L3.53314 4.50002L0.0344238 1.16905Z"
              fill="#000"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`customfields shoulg match snapshot 2`] = `
<DocumentFragment>
  <div>
    <div
      class="name"
    >
      Дополнительные поля
      <button
        class="small button normal undefined basic"
      >
        <div>
          <svg
            fill="none"
            height="5"
            viewBox="0 0 8 5"
            width="8"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M0.0344238 1.16905L0.732916 0.502381L3.54531 3.1561L6.33428 0.499939L7.03473 1.16647L3.53314 4.50002L0.0344238 1.16905Z"
              fill="#000"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </button>
    </div>
    <div
      class="inline line"
    >
      <div
        class="half value"
      >
        child1Sex
      </div>
      <div
        class="half value"
      >
        <span
          class="neutral"
        >
          Пусто
        </span>
      </div>
    </div>
    <div
      class="inline line"
    >
      <div
        class="half value"
      >
        clearTestClient
      </div>
      <div
        class="half value"
      >
        <span
          class="neutral"
        >
          Пусто
        </span>
      </div>
    </div>
    <div
      class="inline line"
    >
      <div
        class="half value"
      >
        isCompany
      </div>
      <div
        class="half value"
      >
        true
      </div>
    </div>
    <div
      class="inline line"
    >
      <div
        class="half value"
      >
        multiline
      </div>
      <div
        class="half value"
      >
        <div>
          1
        </div>
        <div>
          2
        </div>
      </div>
    </div>
    <div
      class="inline line"
    >
      <div
        class="half value"
      >
        singleline
      </div>
      <div
        class="half value"
      >
        <div>
          56
        </div>
        <div>
          <span
            class="neutral"
          >
            Пусто
          </span>
        </div>
      </div>
    </div>
    <div
      class="inline line"
    >
      <div
        class="half value"
      >
        walletCardLink
      </div>
      <div
        class="half value"
      >
        <span
          class="neutral"
        >
          Пусто
        </span>
      </div>
    </div>
    <div
      class="inline line"
    >
      <div
        class="half value"
      >
        walletCardQrCodeLink
      </div>
      <div
        class="half value"
      >
        <span
          class="neutral"
        >
          Пусто
        </span>
      </div>
    </div>
    <div
      class="inline line"
    >
      <div
        class="half value"
      >
        walletCardSubscription
      </div>
      <div
        class="half value"
      >
        false
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`customfields shoulg match snapshot 3`] = `
<DocumentFragment>
  <div>
    <div
      class="name"
    >
      Дополнительные поля
      <button
        class="small button normal undefined basic"
      >
        <div>
          <svg
            fill="none"
            height="5"
            viewBox="0 0 8 5"
            width="8"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M0.0344238 1.16905L0.732916 0.502381L3.54531 3.1561L6.33428 0.499939L7.03473 1.16647L3.53314 4.50002L0.0344238 1.16905Z"
              fill="#000"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
